enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,251 @@
+network of the worker nodes. This is needed for aggregated API services running on the
+worker nodes.
+
+The cluster version operator running on the management cluster connects to the Kubernetes
+API server via the service network and starts applying manifests for SLOs that must run
+on the worker nodes. These include (among others):
+
+- network (dns, multus)
+- ingress
+- image registry
+- OLM
+- samples
+- console
+
+The CVO needs to skip manifests that instantiate control plane operators, machine API related
+operators, and others that do not apply to this topology. These include:
+
+- cluster-version-operator
+- cluster-kube-apiserver-operator
+- cluster-kube-controller-manager-operator
+- cluster-kube-scheduler-operator
+- cluster-openshift-apiserver-operator
+- cluster-openshift-controller-manager-operator
+- cluster-autoscaler-operator
+- cluster-machine-approver
+- insights-operator
+- machine-config-operator
+- cloud-credential-operator
+- cluster-authentication-operator
+
+Components/Changes needed for this to work:
+
+#### OpenShift ROKS Toolkit CLI
+
+CLI tool that takes a release image and a configuration file as input and produces a set
+of manifests that provision a control plane that runs on an existing Kubernetes cluster.
+PKI certificates are not generated by the tool. Manifests of control plane components 
+expect certificates to have been generated before provisioning and to exist as secrets
+and configmaps on the control plane namespace following a predefined contract.
+
+This tool is invoked by IBM Public Cloud provisioning code to when instantiating a new cluster.
+
+#### Control Plane Operator
+An operator that:
+- Projects global configuration into the target cluster. Watches
+  and overwrites any changes to global configuration initiated inside the user
+  cluster.
+- Ensures that the Kubernetes Controller Manager CA bundle includes self-signed CAs
+  generated by the router and the service CA operator.
+
+#### Cluster Version Operator Changes
+In phase I the cluster version operator excludes operator manifests that
+have an annotation in the form:
+
+`exclude.release.openshift.io/[identifier]=true`
+
+whenever a global variable in the form
+
+`EXCLUDE_MANIFESTS=[identifier]`
+
+is present. For IBM public cloud, the value of \[identifier\] is 
+`internal-openshift-hosted`
+
+#### Excluded Manifests
+Manifests for the components that should be skipped by the CVO whenever
+the `EXCLUDE_MANIFESTS=internal-openshift-hosted` environment variable is 
+present, should include the annotation to exclude them. In most cases, only the
+minimum set of manifests to allow skipping the component should be annotated.
+However, in the case of the Machine API and Machine Configuration operators, 
+the CRDs that represent machines, machinesets and autoscalers should also be
+skipped. Monitoring alerts for components that do not get installed in the user
```

Comment:
This part won't be that easy, unless we get provided with a list of it. Also it will not fire if there are no metrics for those components, so don't think its a problem.

Reply:
This is what we addressed with https://github.com/openshift/cluster-monitoring-operator/pull/705
No other alerts related to control plane components have surfaced.